src/mlpack/methods/reinforcement_learning/q_learning_impl.hpp

```diff
@@ -161,8 +226,24 @@ double QLearning<
   // Learn form experience.
   arma::mat gradients;
   learningNetwork.Backward(target, gradients);
+
+  if (prioritized_replay)
+  {
+    gradients = arma::mean(weights) * gradients;
+  }
+
   updater.Update(learningNetwork.Parameters(), config.StepSize(), gradients);
 
+  if (prioritized_replay)
+  {
+    prioritizedReplayMethod.Update(target, sampledActions,
```

Comment:
Do you think we could move this before the ``updater.update`` step? That way we could pass the gradient to the ``replayMethod.Update`` method and do the gradient step inside the ``replayMethod.Update`` method itself. If not we could introduce a new method something like ``GradientUpdate`` that just does nothing for the random replay method. Let me know what you think.

Reply:
Yes, you are more considerate. I agree with you, thanks.

Reply:
> Do you think we could move this before the `updater.update` step? That way we could pass the gradient to the `replayMethod.Update` method and do the gradient step inside the `replayMethod.Update` method itself. If not we could introduce a new method something like `GradientUpdate` that just does nothing for the random replay method. Let me know what you think.

If we pass the gradient to the `replayMethod.Update` method and do the gradient step inside the replayMethod.Update method, should we add the same interface for random replay?

Reply:
I think that is a good idea.

Reply:
Hi, @zoq, I didn't create another new policy class, just remove the two parameters `sampledIndices` and `weights` to `PrioritizedReplay` as class attribution. It seems that more convenient.  So we can pass the same parameter to `sample` function. 

I also move the line which used to update the gradient into the `PrioritizedReplay` class's function `Update(...)`.
```
  gradients = arma::mean(weights) * gradients;
```

Reply:
That sounds reasonable to me.